MzLibUtil/SystemInfo.cs

```diff
@@ -33,250 +28,189 @@ public static string SystemProse()
+        /// // see https://github.com/NimaAra/Easy.Common/blob/master/Easy.Common/DiagnosticReport/DiagnosticReport.cs
+        /// </summary>
+        private static string GetProcessorName()
         {
-            try
-            {
-                return ("Windows version " + Environment.OSVersion.Version.ToString());
-            }
-            catch
-            {
-                return "UNDETERMINED OPERATING SYSTEM";
-            }
-        }
+            string processorName = "UNKNOWN";
 
-        private static string DotNet()
-        {
-#if ONLYNETSTANDARD
-            return "Windows .Net Version could not be determined.\n";
-#else
             try
             {
-                const string subkey = @"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\";
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    var key = Registry.LocalMachine.OpenSubKey(@"HARDWARE\DESCRIPTION\System\CentralProcessor\0\");
+                    processorName = key?.GetValue("ProcessorNameString").ToString() ?? "Not Found";
+                }
 
-                using (RegistryKey ndpKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32).OpenSubKey(subkey))
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
                 {
-                    if (ndpKey != null && ndpKey.GetValue("Release") != null)
-                    {
-                        return ".NET Framework Version: " + CheckFor45PlusVersion((int)ndpKey.GetValue("Release"));
-                    }
-                    else
+                    const string CPUFile = "/proc/cpuinfo";
+                    var cpuLine = File.ReadLines(CPUFile)
+                        .FirstOrDefault(l => l.StartsWith("model name", StringComparison.InvariantCultureIgnoreCase));
+
+                    if (cpuLine != null)
                     {
-                        return ".NET Framework Version 4.5 or later is not detected.";
+                        const string Separator = ": ";
+                        var startIdx = cpuLine.IndexOf(Separator, StringComparison.Ordinal) + Separator.Length;
+                        processorName = cpuLine.Substring(startIdx, cpuLine.Length - startIdx);
                     }
                 }
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    processorName = AsBashCommand("sysctl -n machdep.cpu.brand_string").TrimEnd();
+                }
             }
             catch
             {
-                return "Windows .Net Version could not be determined.\n";
+
             }
-#endif
+
+            return processorName;
         }
 
-        private static string InstalledRam()
+        /// <summary>
+        /// see https://github.com/NimaAra/Easy.Common/blob/master/Easy.Common/DiagnosticReport/DiagnosticReport.cs
+        /// </summary>
+        private static string AsBashCommand(string command)
```

Comment:
pretty simple but would a unit test for this be good?

Reply:
This function only gets run on OSX, so on other machines the unit test would fail (including on AppVeyor). It might work on Linux. I don't think it's easily unit-testable

Reply:
You could test that it fails if the environment is windows